Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,321 @@
+#include "Projectiles.h"
+
+#include "../Helpers.h"
+#include "Mob.h"
+
+#include "../Game.h"
+
+#include <assert.h>
+
+Projectile::Projectile(size_t id) : IOnMapObject(id)
+{
+    lastMove = 0;
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    pixSpeed = 1;
+    dMove = D_DOWN;
+    anchored = false;
+    damage = 0;
+
+    v_level = 5;
+    SetSprite("icons/projectiles.dmi");
+}
+
```

Comment:
I am not sure, but if this code is not temporary because of some local needs (I mean, functions `TryMove`, `CheckMoveTime`) then you should remove them, because they are the same as functions in the base classes.